### PR TITLE
【Hackathon 6th No.29】为 paddle.nn.functional.kl_div 进行功能增强 -part

### PR DESCRIPTION
**Pull request by NKNaN**
### PR Category

User Experience

### PR Types

Improvements

### Description

paddle.nn.functional.kl_div 增加参数 log_target
https://github.com/PaddlePaddle/Paddle/pull/62986

### CODE REVIEW

File: test/deprecated/legacy_test/test_kldiv_loss_op.py [hunk 1]
```diff
@@ -64,34 +70,45 @@ def test_check_grad(self):
...
 
 
 class TestKLDivLossOp4(TestKLDivLossOp):
     def initTestCase(self):
         self.x_shape = (5, 20)
         self.reduction = 'sum'
+        self.log_target = False
+
+
+class TestKLDivLossOp5(TestKLDivLossOp):
```
Comment (zhwesky2010) [inline test/deprecated/legacy_test/test_kldiv_loss_op.py:97]: log_target为True时，再加一个不同shape的测试case吧

File: test/xpu/test_kldiv_loss_op_xpu.py [hunk 2]
```diff
@@ -83,30 +89,40 @@ def test_check_grad(self):
...
             self.reduction = 'none'
+            self.log_target = False
 
     class TestKLDivLossOp4(TestKLDivLossOp):
         def initTestCase(self):
             self.x_shape = (5, 20)
             self.reduction = 'none'
+            self.log_target = False
+
+    class TestKLDivLossOp5(TestKLDivLossOp):
```
Comment (zhwesky2010) [inline test/xpu/test_kldiv_loss_op_xpu.py:112]: 同上

File: python/paddle/nn/functional/loss.py [hunk 3]
```diff
@@ -1649,6 +1655,7 @@ def kl_div(input, label, reduction='mean', name=None):
             if `reduction` is ``'sum'``, the reduced sum loss is returned;
             if `reduction` is ``'none'``, no reduction will be applied.
             Default is ``'mean'``.
+        log_target (bool, optional): Indicate whether `label` is passed in log space. Default is False.
```
Comment (zhwesky2010) [inline python/paddle/nn/functional/loss.py:1658]: 下面的示例代码，可以加一个log_target=True时的代码

File: python/paddle/nn/layer/loss.py [hunk 4]
```diff
@@ -1054,6 +1060,7 @@ class KLDivLoss(Layer):
             if `reduction` is ``'sum'``, the reduced sum loss is returned;
             if `reduction` is ``'none'``, no reduction will be applied.
             Default is ``'mean'``.
+        log_target (bool, optional): Indicate whether `label` is passed in log space. Default is False.
```
Comment (zhwesky2010) [inline python/paddle/nn/layer/loss.py:1063]: 下面的示例代码，可以加一个log_target=True时的代码

File: python/paddle/nn/functional/loss.py (hunk 3 above)
Comment (NKNaN) [reply python/paddle/nn/functional/loss.py:1658]: 已修改

File: python/paddle/nn/layer/loss.py (hunk 4 above)
Comment (NKNaN) [reply python/paddle/nn/layer/loss.py:1063]: 已修改

File: test/deprecated/legacy_test/test_kldiv_loss_op.py (hunk 1 above)
Comment (NKNaN) [reply test/deprecated/legacy_test/test_kldiv_loss_op.py:97]: 已增加

File: test/xpu/test_kldiv_loss_op_xpu.py (hunk 2 above)
Comment (NKNaN) [reply test/xpu/test_kldiv_loss_op_xpu.py:112]: 已增加

**Review by zhwesky2010 — DISMISSED** [review]
LGTM

**Comment by zhwesky2010** [comment]
@NKNaN PR-CI-Static-Check显示示例代码有问题，需要再看一下
infoflow 2024-04-26 15-59-53

**Comment by NKNaN** [comment]
> @NKNaN PR-CI-Static-Check显示示例代码有问题，需要再看一下 

已修改

**Review by zhwesky2010 — APPROVED** [review]
LGTM

**Review by zyfncg — APPROVED** [review]

**Review by sunzhongkai588 — APPROVED** [review]
LGTM for docs
